src/tree-selector/treeSelectorComponent.ts

```diff
@@ -0,0 +1,54 @@
+import * as ng from 'angular';
+
+export class TreeSelectorController {
+
+  public data: any;
+  public selectables: any;
+  public display: any;
+  private getSelected: any;
+  private getData: any;
+
+  /*@ngInject*/
+  constructor(private $http, private $timeout : ng.ITimeoutService) {}
+
+  public $onInit() {
+    this.data = this.parseSelectable(this.data);
+  }
+
+  public lazyLoad = (node) => {
+    const dataPromise = this.getData(node.key);
+    return dataPromise.then((data) => {
+      return this.parseSelectable(data);
+    });
+  }
+
+  public parseSelectable = (data: any) => {
+    return data.map((node, key) => {
+      const parsedData = {...node};
+      if(parsedData.nodes) {
+        parsedData.nodes = this.parseSelectable(parsedData.nodes);
+      }
+      parsedData.selectable = !!parsedData.key.match(this.selectables);
+      return parsedData;
+    });
+  }
+
+  public handleSelect = (data) => {
+    this.$timeout(() => this.getSelected(data));
+    this.display(false);
+  }
+
+}
+
+export default class TreeSelector implements ng.IComponentOptions {
+  public controller = TreeSelectorController;
+  public controllerAs = 'vm';
+  public template = require('./treeSelector.html');
+  public bindings: any = {
+    data: '<',
+    selectables: '@',
+    display: '<',
+    getSelected: '<',
+    getData: '<'
```

Comment:
Our :heart: for consistency will never fade, please express it by naming these the same as in the `tree-view` component:
`onSelect` and `lazyLoad`.